tests/unittests/BackendTest.cpp

```diff
@@ -295,8 +302,43 @@ TEST_P(BackendTest, debugPrint) {
   IRBuilder(IR.get()).createDebugPrintInst("print", *IR->getWeights().begin());
 
   auto function = backend->compileIR(std::move(IR));
-  EE_.insertCompiledFunction("main", std::move(function));
-  EE_.run(*ctx.get());
+
+  // Since we are compiling IR by hand we cannot go through the normal EE route.
```

Comment:
Not for this diff, but I could see an interface where we add a CompiledFunction to the HostManager being useful (use the single node dag builder in Partitioner, etc). That would have made this change really easy.

Reply:
That's a good idea, that way even if we aren't testing the compilation path, we would still be sharing a comment execute path. Wanna write up an issue to capture the thought?